lighting/raymarch/volume.glsl

```diff
@@ -1,103 +1,137 @@
+    - RAYMARCH_VOLUMETRIC_SHADOWS
+    - RAYMARCH_VOLUME_DITHER        0.1
+    - RAYMARCH_ENERGY_CONSERVING
+    - LIGHT_COLOR                   vec3(0.5, 0.5, 0.5)
+    - LIGHT_INTENSITY               1.0
+    - LIGHT_POSITION
+    - LIGHT_DIRECTION
 examples:
     - /shaders/lighting_raymarching_volume.frag
+license: MIT License (MIT) Copyright (c) 2024 Shadi EL Hajj
 */
 
 #ifndef LIGHT_COLOR
 #if defined(GLSLVIEWER)
 #define LIGHT_COLOR u_lightColor
 #else
-#define LIGHT_COLOR vec3(0.5)
+#define LIGHT_COLOR vec3(0.5, 0.5, 0.5)
 #endif
 #endif
 
-#ifndef RAYMARCH_BACKGROUND
-#define RAYMARCH_BACKGROUND vec3(0.0)
+#ifndef LIGHT_INTENSITY
+#define LIGHT_INTENSITY 1.0
 #endif
 
-#ifndef RAYMARCH_SAMPLES
-#define RAYMARCH_SAMPLES 64
+#ifndef RAYMARCH_VOLUME_SAMPLES
+#define RAYMARCH_VOLUME_SAMPLES 64
 #endif
 
-#ifndef RAYMARCH_MIN_DIST
-#define RAYMARCH_MIN_DIST 1.0
+#ifndef RAYMARCH_VOLUME_SAMPLES_LIGHT
+#define RAYMARCH_VOLUME_SAMPLES_LIGHT 32
 #endif
 
-#ifndef RAYMARCH_MAX_DIST
-#define RAYMARCH_MAX_DIST 10.0
+#ifndef RAYMARCH_VOLUME_MAP_FNC
+#define RAYMARCH_VOLUME_MAP_FNC raymarchVolumeMap
 #endif
 
-#ifndef RAYMARCH_MAP_FNC
-#define RAYMARCH_MAP_FNC(POS) raymarchMap(POS)
+#ifndef RAYMARCH_VOLUME_DITHER
+#define RAYMARCH_VOLUME_DITHER 0.1
 #endif
 
 #ifndef FNC_RAYMARCH_VOLUMERENDER
 #define FNC_RAYMARCH_VOLUMERENDER
 
-vec4 raymarchVolume( in vec3 rayOrigin, in vec3 rayDirection, vec3 cameraForward,
-                     out float eyeDepth, out vec3 worldPos, out vec3 worldNormal) {
-
-    const float tmin        = RAYMARCH_MIN_DIST;
-    const float tmax        = RAYMARCH_MAX_DIST;
-    const float fSamples    = float(RAYMARCH_SAMPLES);
-    const float tstep       = tmax/fSamples;
-    const float absorption  = 100.;
-
-    #ifdef LIGHT_POSITION
-    const int   nbSampleLight   = 6;
-    const float fSampleLight    = float(nbSampleLight);
-    const float tstepl          = tmax/fSampleLight;
-    vec3 sun_direction          = normalize( LIGHT_POSITION );
-    #endif
-
-    float T = 1.;
-    float t = tmin;
-    vec4 col = vec4(0.0);
-    vec3 pos = rayOrigin;
-    for(int i = 0; i < RAYMARCH_SAMPLES; i++) {
-        Material res    = RAYMARCH_MAP_FNC(pos);
-        float density = (0.1 - res.sdf);
-        if (density > 0.0) {
-            float tmp = density / fSamples;
-            T *= 1.0 - tmp * absorption;
-            if( T <= 0.001)
-                break;
-
-            col += res.albedo * fSamples * tmp * T;
-                
-            //Light scattering
-            #ifdef LIGHT_POSITION
-            float Tl = 1.0;
-            for (int j = 0; j < nbSampleLight; j++) {
-                float densityLight = RAYMARCH_MAP_FNC( pos + sun_direction * float(j) * tstepl ).sdf;
-                if (densityLight>0.)
-                    Tl *= 1. - densityLight * absorption/fSamples;
-                if (Tl <= 0.01)
-                    break;
+vec3 shadowTransmittance(vec3 position, vec3 rayDirectionL, float stepSizeL) {
```

Comment:
SUPER NIT: just flagging that this break the design rule of filename == function name. But we break and bend that rule all the time. I think is ok. Just thinking to planting the question... on how we can keep that rule more often that not... and how to make consistent the tacit rule about function names that incorporate the previous folder

Reply:
@patriciogonzalezvivo interesting question. I agree we need rules for see.
The way I think of this is private vs. public functions:
* If the function is meant to be public (re-usable elsewhere), then we should def stick to `filename == function`.
* If the function is "private", as in this case (meant to be used as a helper function only called from functions inside the same file) then filename == function` does not apply.
* If you think the above rules make sense, then should be flag private function with a suffix? eg. `_shadowTransmittance`. Just a mere suggestion. I also think it can stay as it is, and the fact it's private can be deduced from `filename != function`

Reply:
Great point. I have been amending that principle often incorporating the "legal name" before the private function. In this case
```
raymarchVolumeShadowTransmittance()
```

The idea behind is that collisions become statistically harder, while the affordance of the function name less important because of their private nature

Reply:
again.. all this is very NIT picking

Reply:
`raymarchVolumeShadowTransmittance()` makes total sense. Let's go with that.
Amended and pushed.